alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -127,16 +132,50 @@ abstract class MonitorRunner {
         monitorCtx: MonitorRunnerExecutionContext
     ): NotificationActionConfigs {
         var destination: Destination? = null
-        val channel: NotificationConfigInfo? = getNotificationConfigInfo(monitorCtx.client as NodeClient, action.destinationId)
+        var notificationPermissionException: Exception? = null
+
+        var channel: NotificationConfigInfo? = null
+        try {
+            channel = getNotificationConfigInfo(monitorCtx.client as NodeClient, action.destinationId)
+        } catch (e: OpenSearchSecurityException) {
+            notificationPermissionException = e
+        }
 
         // If the channel was not found, try to retrieve the Destination
         if (channel == null) {
             destination = try {
-                AlertingConfigAccessor.getDestinationInfo(monitorCtx.client!!, monitorCtx.xContentRegistry!!, action.destinationId)
+                val table = Table(
+                    "asc",
+                    "destination.name.keyword",
+                    null,
+                    1,
+                    0,
+                    null
+                )
+                val getDestinationsRequest = GetDestinationsRequest(
+                    action.destinationId,
+                    0L,
+                    null,
+                    table,
+                    "ALL"
+                )
+
+                val getDestinationsResponse: GetDestinationsResponse = monitorCtx.client!!.suspendUntil {
+                    monitorCtx.client!!.execute(GetDestinationsAction.INSTANCE, getDestinationsRequest, it)
+                }
+                getDestinationsResponse.destinations.firstOrNull()
```

Comment:
Should we just move this to `AlertingConfigAccessor` in case it needs to be used anywhere else?

Reply:
Yes that makes sense. I will take note of this in a future PR.